ci/tasks/comment-on-pr-with-pkl-pipeline-diffs.yml

```diff
@@ -0,0 +1,187 @@
+---
+platform: linux
+image_resource:
+  type: registry-image
+  source:
+    repository: governmentdigitalservice/pay-concourse-runner
+params:
+  GITHUB_TOKEN:
+  CONCOURSE_TEAM:
+  FLY_PASSWORD:
+inputs:
+  - name: pay-ci
+  - name: pkl-pipeline-pr
+outputs:
+  - name: diffs
+run:
+  path: ash
+  args:
+    - -c
+    - |
+      set -euo pipefail
+
+      apk add --no-cache --no-progress github-cli
+
+      export MASTER="pay-ci"
+      export PR="pkl-pipeline-pr"
+
+      GITHUB_PR_URL=$(jq -r '.[] | select(.name == "url") | .value' < "$PR/.git/resource/metadata.json")
+
+      mkdir -p "diffs/${CONCOURSE_TEAM}/"
+
+      FAILED_EVALUATIONS_TMPFILE=$(mktemp)
+
+      for DIR in "$MASTER" "$PR"; do
+        echo "Generting YML from PKL in $DIR"
+
+        if [ ! -d "$DIR/ci/pkl-pipelines/${CONCOURSE_TEAM}" ]; then
+          mkdir -p "$DIR/ci/pkl-pipelines/${CONCOURSE_TEAM}"
+        fi
+
+        cd "$DIR/ci/pkl-pipelines/${CONCOURSE_TEAM}" >> /dev/null 2>&1
+
+        if [ "$(find . -mindepth 1 -maxdepth 1 -type f -name '*.pkl' | wc -l)" -gt 0 ]; then
+          find . -mindepth 1 -maxdepth 1 -type f -name '*.pkl' | while read -r PKL_FILE; do
+            echo "Generating YML for $PKL_FILE"
+            YAML_FILE=$(echo "$PKL_FILE" | sed -E 's/\.pkl$/.yml/')
+            if ! pkl eval "$PKL_FILE" > "$YAML_FILE"; then
+              echo "$PKL_FILE" >> "$FAILED_EVALUATIONS_TMPFILE"
+            fi
+          done
+        fi
+
+        cd - >> /dev/null 2>&1
+      done
+
+      if [ "$(cat "$FAILED_EVALUATIONS_TMPFILE" | wc -l)" -gt 0 ]; then
```

Comment:
UUOC! :)

Reply:
TIL: UUOC acronym :) 

Remove it as I move the whole file to a separate script file :)